src/api/legacy.rs

```diff
@@ -502,6 +505,41 @@ async fn get_usb_mode(bmc: &BmcApplication) -> impl Into<LegacyResponse> {
     )
 }
 
+async fn set_cooling_info(query: Query) -> LegacyResult<()> {
+    let device_str = query
+        .get("device")
+        .ok_or(LegacyResponse::bad_request("Missing `device` parameter"))?;
+    let speed_str = query
+        .get("speed")
+        .ok_or(LegacyResponse::bad_request("Missing `speed` parameter"))?;
+
+    // check if the speed is a number
+    let speed = u8::from_str(speed_str)
+        .map_err(|_| LegacyResponse::bad_request("Parameter `speed` must be a number between 0-255"))?;
```

Comment:
Here I'm just stating the allowed range for `u8`, but if I query the devices first, I could format this to include `device.max_speed` instead of 255. What do you think?

And also, I'm not sure if some device could use `u16`? If this is specified somewhere in some cooling devices' specification, it would be great to know so we can match the type 😜

Reply:
I think it would be relatively low hanging fruit to do a query to get the max_state. You made such a nice code, lets finish it :)

Regarding the data-type, Ive looked-up the driver, and curr/max state is defined as a `unsigned long` . So the correct datatype to store speed in is `std::ffi::c_ulong` i assume.

Reply:
Awesome! I just pushed changes for both the error message and the datatype change. I'm waiting for https://github.com/barrenechea/BMC-Firmware/actions/runs/8599946508 to build and test the changes E2E on my board 😄

Reply:
Everything seems good! Responses are as expected:
```
https://turingpi.local/api/bmc?opt=set&type=cooling&device=cooling_device0&speed=11

{
  "response": [
    {
      "result": "Parameter `speed` must be a number between 0-10"
    }
  ]
}
```

The same is true for setting the speed to `-1` or `qwerty`